src/averageAges.js

```diff
@@ -55,7 +69,18 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(person => (
+    onlyWithSon
+      ? people.find(mother => mother.name === person.mother)
+        && person.sex === 'm'
+      : people.find(mother => mother.name === person.mother)
+  ));
+  const calculateAge = children.map(person =>
+    (person.born - people.find(mother => mother.name === person.mother).born));
+  const calculateAverageAge = calculateAge.reduce((sum, age) => (
+    (sum + age))) / children.length;
```

Comment:
I think u already understand what I want))

Reply:
Thanks for the review! I'll fix it now